Assets/Scripts/Enviroment/EnvironmentBehaviour.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnvironmentBehaviour : MonoBehaviour
+{
+    [Header("Stats")]
+    public float moveSpeed;
+    public float lifeSpan;
+
+    [Header("Despawn when Touching:")] 
+    public bool tree;
+    public bool bush;
+    public bool rock;
+    public bool squiggle;
+    public bool water;
+    
+    private float age;
+    
+    // Update is called once per frame
+    void Update()
+    {
+        Move();
+        Despawn();
+    }
+
+    void Move()
+    {
+        transform.position += Vector3.down * moveSpeed;
+    }
+    
+    void Despawn()
+    {
+        age += Time.deltaTime;
+
+        if (age >= lifeSpan)
+        {
+            Destroy(gameObject);
+        }
+    }
+    
+    private void OnTriggerEnter2D(Collider2D col)
+    {
+        if (CompareTag("Tree") && col.CompareTag("Tree") && tree)
+        {
+            if (age <= 0.5f)
+            {
+                Destroy(gameObject);
+            }
+        }
+        if (CompareTag("Bush") && col.CompareTag("Bush") && bush)
+        {
+            Destroy(gameObject);
+        }
+        if (CompareTag("Rock") && col.CompareTag("Rock") && rock)
+        {
+            Destroy(gameObject);
+        }
+        if (CompareTag("Squiggle") && col.CompareTag("Squiggle") && squiggle)
+        {
+            Destroy(gameObject);
+        }
+        if (CompareTag("Water") && col.CompareTag("Water") && water)
+        {
+            Destroy(gameObject);
+        }
```

Comment:
Gebruik je hier de standaard unity tag system? Zo ja, waarom, we hebben een betere tag system in het project.

Reply:
ja, omdat het eventjes makkelijk was om te gebruiken